src/sentry/api/serializers/models/auth_provider.py

```diff
@@ -1,14 +1,27 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
 from django.db.models import F
 
 from sentry.api.serializers import Serializer, register
 from sentry.models import AuthProvider, Organization, OrganizationMember
 from sentry.utils.http import absolute_uri
 
+if TYPE_CHECKING:
+    from sentry.services.hybrid_cloud.auth import ApiAuthProvider
+
 
 @register(AuthProvider)
 class AuthProviderSerializer(Serializer):
-    def serialize(self, obj, attrs, user):
-        organization = obj.organization
+    def serialize(
+        self,
+        obj: AuthProvider | ApiAuthProvider,
+        attrs,
+        user,
+        organization: Organization | None = None,
```

Comment:
The main change was allowing an organization object to be passed in and serialized inline.  This is important because the auth provider serializer runs in the control silo, thus the relation from obj -> organization does not exist in that mode.  In the future, the typing will be clarified further to `ApiOrganization`, but it is unsafe to do this whilst other code paths do in fact still act on `Organization` (but are not in stable=True test mode yet).